cdc/log.cc

```diff
@@ -539,12 +539,109 @@ class transformer final {
         }
     }
 
+    template<typename Func>
+    static void map_for_each(const bytes_opt& bo, Func&& f) {
+        if (bo) {
+            bytes_view bv(*bo);
+            auto size = read_collection_size(bv, cql_serialization_format::internal());
+            while (size--) {
+                auto k = read_collection_value(bv, cql_serialization_format::internal());
+                auto v = read_collection_value(bv, cql_serialization_format::internal());
+                f(k, v);
+            }
+        }
+    }
+    template<typename Func>
+    static void set_for_each(const bytes_opt& bo, Func&& f) {
+        if (bo) {
+            bytes_view bv(*bo);
+            auto size = read_collection_size(bv, cql_serialization_format::internal());
+            while (size--) {
+                f(read_collection_value(bv, cql_serialization_format::internal()));
+            }
+        }
+    }
+    template<typename Func>
+    static void udt_for_each(const bytes_opt& bo, Func&& f) {
+        if (bo) {
+            bytes_view bv(*bo);
+            std::for_each(tuple_deserializing_iterator::start(bv), tuple_deserializing_iterator::finish(bv), std::move(f));
+        }
+    }
+    static bytes merge_maps(const abstract_type& type, const bytes_opt& prev, const bytes_opt& next, const bytes_opt& deleted) {
```

Comment:
It would probably be more kosher to take `const map_type_impl&` here.
Also, this could then be just called `merge` cuz the `_maps` suffix would be unnecessary due to the type.

Same comment applies to `merge_sets` below.

Reply:
The type is the key type, not the map.